draft-ietf-webtrans-http3.md

```diff
@@ -471,20 +471,25 @@ WebTransport over HTTP/3 uses two different mechanisms to negotiate versions for
 the different parts of the draft.
 
 The hop-by-hop wire format aspects of the protocol are negotiated by changing
-the codepoint used for the SETTINGS_ENABLE_WEBTRANSPORT parameter.  Because of
-that, any WebTransport endpoint MUST wait for the peer's SETTINGS frame before
-sending or processing any WebTransport traffic.  When multiple versions are
-supported by both of the peers, the most recent version supported by both is
-selected.
+the codepoint used for the SETTINGS_ENABLE_WEBTRANSPORT or
+SETTINGS_WEBTRANSPORT_MAX_SESSIONS  parameter.  Because of that, any
+WebTransport endpoint MUST wait for the peer's SETTINGS frame before sending or
+processing any WebTransport traffic.  When multiple versions are supported by
+both of the peers, the most recent version supported by both is selected.
+
+If SETTINGS_WEBTRANSPORT_MAX_SESSIONS is used, and the client supports any
+versions other than the final RFC one, the client MUST send
+SETTINGS_WEBTRANSPORT_MAX_SESSIONS with a positive value (the exact value has
+no semantics).
 
 The data exchanged over the CONNECT stream is transmitted across intermediaries,
 and thus cannot be versioned using a SETTINGS parameter.  To indicate support
 for different versions of the protocol defined in this draft, the clients SHALL
 send a header for each version of the draft supported.  The header
 corresponding to the version described in this draft is
-`Sec-Webtransport-Http3-Draft02`; its value SHALL be `1`.  The server SHALL
+`Sec-Webtransport-Http3-Draft06`; its value SHALL be `1`.  The server SHALL
```

Comment:
We have to manually bump the SETTINGS codepoint either way.

Actually, should we just keep it at draft02?  As far as I can tell, we've not made incompatible changes between capsules used in 02 and 06.

Reply:
I think tying this to draft version is maybe less fun than tying it to "wire format compatibility version", although I don't feel super strongly either way. It's certainly confusing if it's the latter but named as the former.

Reply:
Keeping header to `draft02` for now.

Reply:
Silly question: what's the purpose of this header if it's fixed at draft02?

Reply:
The idea was that if we make an incompatible version to capsules, we could use this as an escape hatch (since capsules are transmitted E2E, whereas settings are hop-by-hop).

Reply:
Given that we only have a single capsule (CLOSE_WEBTRANSPORT_SESSION) I'd say this complexity isn't worth it and I'd remove the header entirely, because at this point it's very likely to cause confusion by saying draft02 when we're actually using a different version of the draft that is incompatible with draft02

Reply:
If we have a setting that changes as we revise things, I'm not sure that this provides value now.  The real tricky stuff that might change between revisions (and any ultimate RFC) seems to be captured by that setting.